clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -261,26 +287,29 @@ private boolean renameDirectory(Path src, Path dst) throws IOException {
             }
             // lakefsFs only has non-empty directories. Therefore, if the destination is an existing directory we consider
             // it to be non-empty. The behaviour is same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1530
-            LOG.error("renameDirectory: rename src {} to dst {}: dst is a non-empty directory.", src, dst);
-            return false;
+            if (!dstFileStatus.isEmptyDirectory()) {
```

Comment:
Not sure I agree. 
If I rename "a/" to "b/" and "b/" already exists, I think it should fail  - even if it is empty.

Reply:
update the code to address the above - thanks!